packages/caliper-core/lib/worker/tx-observers/prometheus-manager-tx-observer.js

```diff
@@ -0,0 +1,97 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const TxObserverInterface = require('./tx-observer-interface');
+
+const PrometheusUpdateMessage = require('../../common/messages/prometheusUpdateMessage');
+
+/**
+ * Prometheus Manager TX observer used to send updates to the Prometheus scrape server in the manager.
+ */
+class PrometheusManagerTxObserver extends TxObserverInterface {
+    /**
+     * Initializes the instance.
+     * @param {object} options The observer configuration object.
+     * @param {MessengerInterface} messenger The worker messenger instance. Not used.
+     * @param {number} workerIndex The 0-based index of the worker node.
+     * @param {string} managerUuid The UUID of the manager messenger.
+     */
+    constructor(options, messenger, workerIndex, managerUuid) {
+        super(messenger, workerIndex);
+
+        this.managerUuid = managerUuid;
+    }
+    /**
+     * Called when TXs are submitted.
+     * @param {number} count The number of submitted TXs. Can be greater than one for a batch of TXs.
+     */
+    txSubmitted(count) {
+        const message = new PrometheusUpdateMessage(this.messenger.getUUID(), [this.managerUuid], {
+            event: 'txSubmitted',
+            workerIndex: this.workerIndex,
+            roundIndex: this.currentRound,
+            roundLabel: this.roundLabel,
+            count: count
+        });
+        this.messenger.send(message);
+    }
+
+    /**
+     * Called when TXs are finished.
+     * @param {TxStatus | TxStatus[]} results The result information of the finished TXs. Can be a collection of results for a batch of TXs.
+     */
+    txFinished(results) {
+        if (Array.isArray(results)) {
+            for (const result of results) {
+                // pass/fail status from result.GetStatus()
+                const message = new PrometheusUpdateMessage(this.messenger.getUUID(), [this.managerUuid], {
+                    event: 'txFinished',
+                    workerIndex: this.workerIndex,
+                    roundIndex: this.currentRound,
+                    roundLabel: this.roundLabel,
+                    status: result.GetStatus(),
+                    latency: result.GetTimeFinal() - result.GetTimeCreate()
+                });
+                this.messenger.send(message);
+            }
+        } else {
+            // pass/fail status from result.GetStatus()
+            const message = new PrometheusUpdateMessage(this.messenger.getUUID(), [this.managerUuid], {
+                event: 'txFinished',
+                workerIndex: this.workerIndex,
+                roundIndex: this.currentRound,
+                roundLabel: this.roundLabel,
+                status: results.GetStatus(),
+                latency: (results.GetTimeFinal() - results.GetTimeCreate())/1000
```

Comment:
Here you divide latency by 1000, but above you don't. I assume they should both be the same

Reply:
I had noticed this, but I had left it as such so that it is inline with the implementations in the other prometheus observers:

https://github.com/hyperledger/caliper/blob/5938426a10af642989da0bb4fceeeb185e24e4aa/packages/caliper-core/lib/worker/tx-observers/prometheus-tx-observer.js#L163-L175

https://github.com/hyperledger/caliper/blob/c648fa9510fdb856d9cfec455dc8eb42cee2600b/packages/caliper-core/lib/worker/tx-observers/prometheus-push-tx-observer.js#L165-L177

I would like to know if this is a bug in the codebase or is it intended. If this is a bug, I could fix this for the other observers in a separate PR.

Reply:
That looks like a bug to me in the other observers and both entries should divide by 1000 as it looks like the latency histogram is specified as seconds